### PR TITLE
fix: crash on start if trace-logging and no process.argv[1]

### PR DESCRIPTION
**Pull request by trentm**
Fixes: #2002

### CODE REVIEW

**Comment by trentm** [comment]
Here is what the crash-case looks like with the fix:

```
% ELASTIC_APM_LOG_LEVEL=trace node -r ../start -e 'console.log("hi")'
agent configured correctly { pid: 47398,
  ppid: 46122,
  arch: 'x64',
  platform: 'darwin',
  node: 'v10.23.0',
  agent: '3.12.1',
  ancestors:
   [ '/Users/trentm/el/apm-agent-nodejs2/index.js',
     '/Users/trentm/el/apm-agent-nodejs2/start.js',
     null,
     [length]: 3 ],
  startTrace:
   [ 'at Agent.start (/Users/trentm/el/apm-agent-nodejs2/lib/agent.js:178:11)',
     'at Object.<anonymous> (/Users/trentm/el/apm-agent-nodejs2/start.js:3:32)',
     'at Module._compile (internal/modules/cjs/loader.js:778:30)',
     'at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)',
     'at Module.load (internal/modules/cjs/loader.js:653:32)',
     'at tryModuleLoad (internal/modules/cjs/loader.js:593:12)',
     'at Function.Module._load (internal/modules/cjs/loader.js:585:3)',
     'at Module.require (internal/modules/cjs/loader.js:692:17)',
     'at Module._preloadModules (internal/modules/cjs/loader.js:901:12)',
     'at preloadModules (internal/bootstrap/node.js:602:7)',
     [length]: 10 ],
  main: '<could not determine>',
  dependencies: '<could not determine>',
  conf:
   { ignoreUrlStr: [ [length]: 0 ],
     ignoreUrlRegExp: [ [length]: 0 ],
     ignoreUserAgentStr: [ [length]: 0 ],
     ignoreUserAgentRegExp: [ [length]: 0 ],
     transactionIgnoreUrlRegExp: [ [length]: 0 ],
     sanitizeFieldNamesRegExp:
      [ { /^password$/i [lastIndex]: 0 },
        { /^passwd$/i [lastIndex]: 0 },
        { /^pwd$/i [lastIndex]: 0 },
        { /^secret$/i [lastIndex]: 0 },
        { /^.*key$/i [lastIndex]: 0 },
        { /^.*token.*$/i [lastIndex]: 0 },
        { /^.*session.*$/i [lastIndex]: 0 },
        { /^.*credit.*$/i [lastIndex]: 0 },
        { /^.*card.*$/i [lastIndex]: 0 },
        { /^authorization$/i [lastIndex]: 0 },
        { /^set\x2dcookie$/i [lastIndex]: 0 },
        { /^pw$/i [lastIndex]: 0 },
        { /^pass$/i [lastIndex]: 0 },
        { /^connect\.sid$/i [lastIndex]: 0 },
        [length]: 14 ],
     abortedErrorThreshold: 25,
     active: true,
     apiRequestSize: 786432,
     apiRequestTime: 10,
     asyncHooks: true,
     breakdownMetrics: true,
     captureBody: 'off',
     captureErrorLogStackTraces: 'messages',
     captureExceptions: true,
     captureHeaders: true,
     captureSpanStackTraces: true,
     centralConfig: true,
     cloudProvider: 'auto',
     disableInstrumentations: [ [length]: 0 ],
     environment: 'development',
     errorMessageMaxLength: 2048,
     errorOnAbortedRequests: false,
     filterHttpHeaders: true,
     instrument: true,
     instrumentIncomingHTTPRequests: true,
     logLevel: 'trace',
     logUncaughtExceptions: false,
     metricsInterval: 30,
     metricsLimit: 1000,
     sanitizeFieldNames:
      [ 'password',
        'passwd',
        'pwd',
        'secret',
        '*key',
        '*token*',
        '*session*',
        '*credit*',
        '*card*',
        'authorization',
        'set-cookie',
        'pw',
        'pass',
        'connect.sid',
        [length]: 14 ],
     serverTimeout: 30,
     sourceLinesErrorAppFrames: 5,
     sourceLinesErrorLibraryFrames: 5,
     sourceLinesSpanAppFrames: 0,
     sourceLinesSpanLibraryFrames: 0,
     stackTraceLimit: 50,
     transactionIgnoreUrls: [ [length]: 0 ],
     transactionMaxSpans: 500,
     transactionSampleRate: 1,
     useElasticTraceparentHeader: true,
     usePathAsTransactionName: false,
     verifyServerCert: true,
     serviceName: 'elastic-apm-node',
     serviceVersion: '3.12.1',
     serverHost: 'localhost:8200',
     serverPort: 8200 } }
adding hook to Node.js module loader
hi
shimming http@10.23.0 module
shimming http.Server.prototype.emit function
shimming http.ServerResponse.prototype.writeHead function
shimming http.request function
shimming http.get function
no active transaction found - cannot build new span
intercepted call to http.request { id: null }
no active transaction found - cannot build new span
intercepted call to http.request { id: null }
no active transaction found - cannot build new span
intercepted call to http.request { id: null }
APM Server transport error (ECONNREFUSED): connect ECONNREFUSED 127.0.0.1:8200
no cloud metadata servers responded
no active transaction found - cannot build new span
intercepted call to http.request { id: null }
APM Server transport error (ECONNREFUSED): connect ECONNREFUSED 127.0.0.1:8200
```

File: lib/agent.js [hunk 1]
```diff
@@ -174,12 +174,14 @@ Agent.prototype.start = function (opts) {
     return this
   } else if (this._conf.logLevel === 'trace') {
     var _ancestors = ancestors(module)
-    var basedir = path.dirname(process.argv[1])
     var stackObj = {}
     Error.captureStackTrace(stackObj)
 
+    // Attempt to load package.json from process.argv.
+    var pkg = null
```
Comment (astorm) [inline lib/agent.js:181]: I see we've pulled the `pkg` variable up to the outer scope for readability, and possibly [some side eye at `var` hoisting](https://developer.mozilla.org/en-US/docs/Glossary/Hoisting).

File: lib/agent.js [hunk 2]
```diff
@@ -174,12 +174,14 @@ Agent.prototype.start = function (opts) {
...
     try {
-      var pkg = require(path.join(basedir, 'package.json'))
+      var basedir = path.dirname(process.argv[1] || '.')
+      pkg = require(path.join(basedir, 'package.json'))
```
Comment (astorm) [inline lib/agent.js:184]: I see we're short circuiting to a value of `.` if `process.argv[1]` is not defined.

File: lib/agent.js [hunk 3]
```diff
@@ -191,8 +193,8 @@ Agent.prototype.start = function (opts) {
       agent: version,
       ancestors: _ancestors,
       startTrace: stackObj.stack.split(/\n */).slice(1),
-      main: pkg && pkg.main,
-      dependencies: pkg && pkg.dependencies,
+      main: pkg ? pkg.main : '<could not determine>',
+      dependencies: pkg ? pkg.dependencies : '<could not determine>',
       conf: this._conf.toJSON()
```
Comment (astorm) [inline lib/agent.js:198]: I see when assigning the `main` and `dependencies` properties to the object we're logging that we're using a ternary to provide a default value of `"<could not determine>"` if there was some issue loading `package.json`.  If `package.json` was loaded but `pkg.main` or `pkg.dependencies` is not set then a value of _undefined_ will still be recorded (which seems correct/accurate)

**Review by astorm — APPROVED** [review]
Approved pending CI review

**Pull request merged**